template.tpl

```diff
@@ -186,252 +186,604 @@ ___TEMPLATE_PARAMETERS___
+                      "displayName": "Value",
+                      "simpleValueType": true,
+                      "valueValidators": [
+                        {
+                          "type": "NON_EMPTY"
+                        }
+                      ]
                     },
-                    {
-                      "value": "userSet",
-                      "displayValue": "Constant or Variable"
-                    }
-                  ],
-                  "simpleValueType": true,
-                  "valueValidators": [
-                    {
-                      "type": "NON_EMPTY"
-                    }
-                  ]
-                },
-                "isUnique": false
+                    "isUnique": false
+                  }
+                ],
+                "enablingConditions": [
+                  {
+                    "paramName": "customUseVariables",
+                    "paramValue": false,
+                    "type": "EQUALS"
+                  }
+                ]
               },
               {
-                "param": {
-                  "type": "TEXT",
-                  "name": "value",
-                  "displayName": "Value",
-                  "simpleValueType": true,
-                  "valueValidators": [
-                    {
-                      "type": "NON_EMPTY"
-                    }
-                  ]
-                },
-                "isUnique": false
+                "type": "SIMPLE_TABLE",
+                "name": "customEntitiesFromVar",
+                "displayName": "Set custom context entities through variables",
+                "simpleTableColumns": [
+                  {
+                    "defaultValue": "",
+                    "displayName": "Event Name",
+                    "name": "eventName",
+                    "type": "TEXT",
+                    "valueValidators": [
+                      {
+                        "type": "NON_EMPTY"
+                      }
+                    ],
+                    "isUnique": true
+                  },
+                  {
+                    "defaultValue": "",
+                    "displayName": "Custom Entities Array",
+                    "name": "customContexts",
+                    "type": "TEXT",
+                    "valueValidators": [
+                      {
+                        "type": "NON_EMPTY"
+                      }
+                    ]
+                  }
+                ],
+                "enablingConditions": [
+                  {
+                    "paramName": "customUseVariables",
+                    "paramValue": true,
+                    "type": "EQUALS"
+                  }
+                ]
               }
-            ],
-            "enablingConditions": [
+            ]
+          },
+          {
+            "type": "GROUP",
+            "name": "globalEntitiesGroup",
+            "displayName": "Global context entities",
```

Comment:
I don't think its clear from the Tag UI what a custom entity and a global entity is. The difference in the UI is very subtle (no event name) and I had to read the code before I noticed what the difference was! I think adding in the help text somewhere that helps users understand the difference would be good. I'm not sure help text renders on groups, but maybe you could add a label into the group?

> Custom Entities are attached where the event_name matches the incoming event, whereas global entities will be applied to all events

Reply:
Thank you for reviewing @paulboocock ! Indeed help texts on groups do not seem to render, so added it in a label under a "Context entities settings" group.
This is what it looks like

![context_entities_settings](https://user-images.githubusercontent.com/35838196/202091375-76119291-028e-4bdc-a45d-39609fd3f9bb.png)

Labels render on tag summary as well (no option to disable). Is this ok?

![sp_tag_summary](https://user-images.githubusercontent.com/35838196/202091703-72cd6bb0-153d-4875-9300-ed9eafa0269f.png)

Reply:
Can we try and add the label to the help text of the table, or something lower down the nesting so it doesn't show at this level?

Reply:
Labels insist on showing up at summary level, so just splitted the help text for custom and global and added it to the respective tables' help text:

![select-area_20221122132929](https://user-images.githubusercontent.com/35838196/203304601-64f6b0ff-09c2-4bd3-9ba8-cd332c7968ee.gif)